src/pack.c

```diff
@@ -46,6 +48,12 @@ static int packfile_error(const char *message)
 	return -1;
 }
 
+int git_pack_global_init(void)
+{
+	global_ignore_keep_pack_file = !!getenv("GIT_IGNORE_PACK_KEEP");
```

Comment:
Generally we try to avoid adding new global initializers.  For something where we're just reading an environment variable, and there aren't really any race conditions or otherwise need to be initialized strictly at the start of library execution, we can probably get by with simpler constructs.

A simple `get`ter type function that does the lookup on-demand would probably suffice?

Reply:
The main reason for not calling 'getenv()' each time is to avoid the performance overhead of such calls when the library is used in a server environment. Fetching the value and storing it in a local variable will make it better.

Had we fetched the settings from the environment during the startup, it would have ensured
thread safety - calls to getenv() racing with setenv() [if it exists in main application].

Ideally, I would have preferred to do it using pthread_once. As you rightly pointed, it could be an overkill for this use case.

On a side topic: If we expect customization of the behavior of libgit2, it might be nice to have a config file that we can load at startup and use it across the code/runtime.

Reply:
Hello, and welcome! I'd suggest adding that as a flag to `git_libgit2_opts`, as it would move the responsibility of initialization (races & all) to the user's code. I can see that there's some prior art in the `git_repository__fsync_gitdir` flag.

Reply:
> Hello, and welcome! I'd suggest adding that as a flag to `git_libgit2_opts`, as it would move the responsibility of initialization (races & all) to the user's code. I can see that there's some prior art in the `git_repository__fsync_gitdir` flag.

This is a good approach. Shifting the responsibility to the main application is best for a library.

Reply:
I agree with @tiennou's suggestion to use a settable option here - we try to avoid environment variables when possible, preferring to let the hosting application handle set up.